### PR TITLE
Make LSP method names more consistent

### PR DESCRIPTION
**Pull request by koliyo**
Both internally and against the spec
Also add missing linkedEditingRange and moniker methods

### CODE REVIEW

**Comment by koliyo** [comment]
Ugh, I realize things are not as I expected, the spec has similar, imo, inconsistencies: `textDocument/didChange` vs `DidChangeTextDocumentParams`.

I will update PR to reflect this.

**Comment by koliyo** [comment]
Ok have restored the param names to match spec

File: Sources/LanguageServerProtocol/LanguageFeatures/Moniker.swift [hunk 1]
```diff
@@ -1,3 +1,50 @@
...
+	case document
+	case project
+	case group
+	case scheme
+	case global
+	// public static let document = "document"
+	// public static let project = "project"
+	// public static let group = "group"
+	// public static let scheme = "scheme"
+	// public static let global = "global"
```
Comment (mattmassicotte) [inline Sources/LanguageServerProtocol/LanguageFeatures/Moniker.swift:30]: Are these commented lines intentionally left in?

File: Sources/LanguageServerProtocol/LanguageFeatures/Moniker.swift [hunk 2]
```diff
@@ -1,3 +1,50 @@
...
+}
+
+public enum MonikerKind : Codable, Sendable{
+	case _import
+	case export
+	case local
+	// public static let _import = "import"
+	// public static let export = "export"
+	// public static let local = "local"
```
Comment (mattmassicotte) [inline Sources/LanguageServerProtocol/LanguageFeatures/Moniker.swift:39]: these too?

File: Sources/LanguageServerProtocol/LanguageFeatures/LinkedEditingRange.swift [hunk 3]
```diff
@@ -1,3 +1,26 @@
...
+        self.partialResultToken = partialResultToken
+        self.textDocument = textDocument
+        self.position = position
+    }
+}
+
+public struct LinkedEditingRanges : Codable, Sendable {
+	public let ranges: [LSPRange]
+	public let wordPattern: String?
+}
```
Comment (mattmassicotte) [inline Sources/LanguageServerProtocol/LanguageFeatures/LinkedEditingRange.swift:23]: Should this have an init?

**Review by mattmassicotte — COMMENTED** [review]
Just some notes - I'm 100% fine merging and addressing these things, if needed, separately.

**Comment by mattmassicotte** [comment]
I just had a question about those comments. If those are ok with this, I'm ready to merge!

**Comment by koliyo** [comment]
Remove the unused code, sticking with the enum approach :)

**Review by mattmassicotte — APPROVED** [review]

**Comment by mattmassicotte** [comment]
This is great!

**Pull request merged**